src/main/java/omero/gateway/Gateway.java

```diff
@@ -1154,7 +1154,53 @@ private boolean isSessionID(String s) {
         }
         return true;
     }
-    
+
+    /**
+     * Test if server and client versions are compatible.
+     * Not {@code private} only to allow unit testing.
+     * @param server server version, no {@code null}s
+     * @param client client version, no {@code null}s
+     * @return if they are compatible
+     */
+    boolean isCompatibleVersion(String[] server, String[] client) {
+        if (server.length < 1 || client.length < 1) {
+            return true;
+        }
+        /* Major versions present so compare. */
+        if (!server[0].equals(client[0])) {
+            /* Major version does not match. */
+            return false;
+        }
+        if (server.length < 2 || client.length < 2) {
+            return true;
+        }
+        /* Minor versions present so compare. */
+        if (server[1].equals(client[1])) {
+            /* Minor version matches. */
+            return true;
+        }
+        /* Major version matches, minor version does not match. */
+        try {
+            final int major = Integer.parseUnsignedInt(server[0]);
+            if (major < 5) {
+                /* Server and client are earlier than 5.0 so minor difference is breaking. */
+                return false;
+            } else if (major > 5) {
+                /* Server and client are 6.0 or later so minor difference is not breaking. */
+                return true;
+            }
+            /* Minor difference is breaking only before 5.5. */
+            final int serverMinor = Integer.parseUnsignedInt(server[1]);
+            final int clientMinor = Integer.parseUnsignedInt(client[1]);
+            return serverMinor >= 5 && clientMinor >= 5;
+        } catch (NumberFormatException nfe) {
+            log.warn(this, String.format(
+                    "cannot compare server version %s.%s with client version %s.%s, assuming incompatible",
+                    server[0], server[1], client[0], client[1]));
```

Comment:
Could you print out the whole version instead of just major.minor?

Reply:
Question for others: what should comparing `5.A` with `5.A` return?

Reply:
The method is `isCompatible` so true, no?

Reply:
I *can*, I thought it might be useful to highlight the consequential aspect of the versioning.

Reply:
I think it's more useful for someone investigating a version failure if they see the full strings that were passed to the function, otherwise they may be confused about where the `%s.%s` version comes from.